### ISSUE TITLE
Can I use custom distance as input for UMAP

### ISSUE THREAD

**Issue by LiLin-biosoft** [lmcinnes/umap#348]
I have a nxn distance matrix for n objects. Can I directly use this as input rather than letting UMAP calculate the distance? The R package UMAP accepts a distance matrix but I cannot find corresponding setting  in python version.  Thank you for your help

**Comment by jay-reynolds** [comment]
Yes, you can.
Set metric='precomputed' and then pass the distance matrix to fit(), transform(), or fit_transform().
